app/views/articles/_v2_form.html.erb

```diff
@@ -103,6 +103,7 @@
   <%= render "pages/editor_liquid_help" %>
   <%= render "pages/editor_frontmatter_help" %>
 </div>
+<div id="i18n-translations" data-translations="<%= { I18n.locale.to_sym => { views: { editor: I18n::JS.translations[I18n.locale.to_sym][:views][:editor] } } }.to_json %>"></div>
```

Comment:
Does this create a second div with the same `id` as the [core translations](https://github.com/forem/forem/blob/main/app/views/layouts/application.html.erb#L110)? Will that work correctly if locale.js calls getElementById (and only sees one of them)?

Does it make sense, if we're going to add translation keys outside the core file, either to enable loading translations from multiple locations (rather than a single `<div id="i18n-translations" ...>`), rather than losing the basic translations already loaded? I'm not entirely sure, but we might want to consider using querySelectorAll and adding a class like `.translation-data` or revisit the front end i18n setup.   

Does this translations div get read after we make a page transition? For example, from the home page /, there's a fetch of /new?i=i when I hover over the button, and when I click the button, the page content is replaced, while the browser doesn't make a normal get request.

I think there's a lot of value in the work you've done in the article views to add translations to strings, this specific implementation might not work correctly, and possibly causes problems with interactions.

Reply:
Even though we're not ready to accept the Portuguese translations, it might help to preview them live, locally. In application_controller.rb there is a [`determine_locale`](https://github.com/forem/forem/blob/main/app/controllers/application_controller.rb#L180-L187) method that allows _per request_ locale setting (overriding the admin controlled default locale setting for the forem) - in order to preview the PT translations you may need to add the "pt" language key to that list: 

```ruby
    I18n.locale = if %w[en fr pt].include?(params[:locale])
                    params[:locale]
                  else
                    Settings::UserExperience.default_locale
                  end
```

That allows viewing the changes, for example the home page http://localhost:3000/locale/pt/ (looks as intended)

![Screenshot from 2022-01-06 09-52-34](https://user-images.githubusercontent.com/1237369/148410772-3b8c85c7-612e-43cc-930e-48688e787610.png)

And the editor page (Create Post) http://localhost:3000/locale/pt/new (looks great, localized strings are used):

![Screenshot from 2022-01-06 09-54-18](https://user-images.githubusercontent.com/1237369/148411012-f4ecdf6c-e2e2-4793-a4a0-4c37657ba9dd.png)

However, when I start from the home page, and navigate to the new article, I get a missing translation (this is I think tied to the front end reading the translations div once on the home page, and not reloading the page during transitions, so only the core: keys are present):

![Screenshot from 2022-01-06 09-56-44](https://user-images.githubusercontent.com/1237369/148411361-44a9bbc5-ec58-490d-9b99-e71dfad2b988.png)

That's the problem I was pointing out, with the way (and when) we're loading the translation info for the front end, that the information available depends on the first page loaded, and subsequent transitions won't add/reload the available translations in the locale file.

A similar problem occurs if I start at the editor page http://localhost:3000/locale/pt/new , and return to the home page by the button (default button text is "DEV(local)"):

![Screenshot from 2022-01-06 09-58-03](https://user-images.githubusercontent.com/1237369/148411693-63151864-ac54-447f-83ed-dac1b6e3c92c.png)